src/formio/components/AddressNL.stories.js

```diff
@@ -0,0 +1,128 @@
+import {expect} from '@storybook/jest';
+import {userEvent, waitFor, within} from '@storybook/testing-library';
+
+import {withUtrechtDocument} from 'story-utils/decorators';
+import {ConfigDecorator} from 'story-utils/decorators';
+import {sleep} from 'utils';
+
+import {
+  mockBRKZaakgerechtigdeInvalidPost,
+  mockBRKZaakgerechtigdeValidPost,
+} from './AddressNL.mocks';
+import {SingleFormioComponent} from './story-util';
+
+export default {
+  title: 'Form.io components / Custom / Address NL',
+  decorators: [withUtrechtDocument, ConfigDecorator],
+  args: {
+    type: 'addressNL',
+    key: 'addressNL',
+    label: 'Address NL',
+    extraComponentProperties: {
+      validate: {
+        required: true,
+      },
+    },
+    evalContext: {},
+  },
+  argTypes: {
+    key: {type: {required: true}},
+    label: {type: {required: true}},
+    type: {table: {disable: true}},
+  },
+};
+
+export const Default = {
+  render: SingleFormioComponent,
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    const postcodeInput = await canvas.findByRole('textbox', {name: 'Postcode'});
+    userEvent.type(postcodeInput, '1234AB');
+
+    const houseNumberInput = await canvas.findByRole('textbox', {name: 'Huis nummer'});
+    userEvent.type(houseNumberInput, '1');
+    userEvent.tab();
+
+    // No errors if the two required fields are filled:
+    let error = canvas.queryByText('Required');
+    await expect(error).toBeNull();
+
+    userEvent.clear(postcodeInput);
+    await sleep(300);
+    userEvent.tab();
+    await sleep(300);
+
+    // Error if postcode not filled:
+    error = await canvas.findByText('Required');
+    await expect(error).not.toBeNull();
+  },
+};
+
+export const WithBRKValidation = {
+  render: SingleFormioComponent,
+  parameters: {
+    msw: {
+      handlers: [mockBRKZaakgerechtigdeValidPost],
+    },
+  },
+  args: {
+    extraComponentProperties: {
+      validate: {
+        required: false,
+        plugins: ['brk-Zaakgerechtigde'],
+      },
+    },
+  },
+  play: async ({canvasElement}) => {
+    const canvas = within(canvasElement);
+
+    const postcodeInput = await canvas.findByRole('textbox', {name: 'Postcode'});
+    await userEvent.type(postcodeInput, '1234AB');
+
+    const houseNumberInput = await canvas.findByRole('textbox', {name: 'Huis nummer'});
+    await userEvent.type(houseNumberInput, '1');
+
+    await sleep(300);
+    await userEvent.tab();
+    await sleep(300);
+
+    await waitFor(async () => {
+      expect(await canvas.queryByText('User is not a zaakgerechtigde for property.')).toBeNull();
+    });
```

Comment:
Probably better to set up a form with a submit handler via `jest.fn()` and assert that is called, as it indicates that there are no validation errors. It takes only a small change in the mock error message returned and this test lost its value :grimacing: 

I think we have something like that elsewhere, I'll see if I can find it tomorrow.